tests/phpunit/integration/Modules/AnalyticsTest.php

```diff
@@ -517,8 +517,10 @@ public function test_tracking_disabled( $settings, $logged_in, $is_tracking_acti
 			// When tracking is active, the `googlesitekit_analytics_tracking_opt_out` action should not be called.
 			$this->assertEquals( 0, did_action( 'googlesitekit_analytics_tracking_opt_out' ) );
 		} else {
-			// When tracking is disabled, the opt out snippet should be present.
-			$this->assertStringContainsString( 'window["ga-disable-UA-21234567-8"] = true', $head_html );
+			if ( ! empty( $settings['propertyID'] ) ) {
+				// When tracking is disabled, the opt out snippet should be present.
+				$this->assertStringContainsString( 'window["ga-disable-UA-21234567-8"] = true', $head_html );
+			}
```

Comment:
Perhaps here we should add an assertion that it does not contain `window["ga-disable-"]`  when there is no propertyID?